src/wallet/walletdb.cpp

```diff
@@ -810,6 +810,7 @@ void ThreadFlushWalletDB(const string& strFile)
                 if (nRefCount == 0)
                 {
                     boost::this_thread::interruption_point();
+                    const std::string& strFile = pwalletMain->strWalletFile;
```

Comment:
In theory, this should only be done when holding cs_wallet. I guess changing CWallet::strWalletFile to a `const std::string` is not possible with the current concept of setting the filename in a instance method.
But maybe I'm wrong.